dev/test-dependencies.sh

```diff
@@ -33,7 +33,7 @@ HADOOP2_MODULE_PROFILES="-Phive-thriftserver -Pmesos -Pkubernetes -Pyarn -Phive"
 MVN="build/mvn"
 HADOOP_PROFILES=(
     hadoop-2.7
-    hadoop-3.1
+    hadoop-3
```

Comment:
Can we use `hadoop-3.2` instead of `hadoop-3`? According to the description, we will have some problem with Hadoop 3.1. `hadoop-3` will give an impression to support Hadoop 3.0 and 3.1 mistakenly.

Reply:
I don't particularly think this is the place to document that level of support. A comment, or documentation, yes. Flipping it around, does a `hadoop-3.2` profile suggest it doesn't work with `hadoop-3.3`? which I expect it will.

Reply:
I expect that the relationship between `hadoop-3.2` and `hadoop-3.3` will be the same with `hadoop-2.6` and `hadoop-2.7`.

Reply:
The fact that there were so many Hadoop profiles is 'legacy'; once upon a time it was pretty important to separate even minor releases around the time of 2.0-2.2. There has long been no point in differentiating 2.6 and 2.7 for example, and shouldn't be going forward. At least I'd prefer to not split this up until there's a need. We really don't want to support versions of Hadoop separately.

Reply:
I agree with you at the following. We should not increase the burden.
> We really don't want to support versions of Hadoop separately.

So, my question is `Apache Spark 3.0 is tested with Hadoop 3.0 and Hadoop 3.1` in the community in Jenkins? This PR will ignore them and move forward to 3.2. That's the reason I prefer `hadoop-3.2`

We can keep `hadoop-3.2` for Hadoop 3.3 (if there is no problem with Hadoop 3.2 library). That is our approach with `hadoop-2.7` profile.

Reply:
I don't feel so strongly about it, but what happens when you test against Hadoop 3.3? the -3.2 profile looks odd. `-Phadoop-3.2 -Dhadoop.version=3.3.0`? If it's not specific to Hadoop 3.2, it just doesn't seem so useful to say it is in the name.

Reply:
@srowen . That's the current Apache Spark way for Hadoop 2.8 and 2.9.
- Building `-Phadoop-2.7 -Dhadoop.version=2.8.5`
- Building `-Phadoop-2.7 -Dhadoop.version=2.9.x`

We had better be consistent.

Reply:
If `-Phadoop-2.7` or `hadoop-3.2` is incompatible with the some versions, we need have another profile explicitly instead of hiding with some ambiguous name like `hadoop-3`.

Reply:
Yes, none of the way it works now is a good example. I won't argue it further, and am neutral on the naming.